src/components/Provider.js

```diff
@@ -0,0 +1,125 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import Mycontext from './Mycontext';
+
+function Provider({ children }) {
+  const [starwars, setStarwars] = useState([]);
+  // Diogo Lemos
+  const [filters, setFilter] = useState({
+    filterByName: {
+      name: '',
+    },
+    filterByNumericValues: [
+      // {
+      //   column: '',
+      //   comparison: '',
+      //   value: '0',
+      // }
+    ],
+  });
+
+  const [click, setClick] = useState(false);
+
+  const [stateFilters] = useState({
+    Population: true,
+    Orbital_period: true,
+    Diameter: true,
+    Rotation_period: true,
+    Surface_water: true,
+  });
+
+  const [option] = useState([
+    'maior que', 'menor que', 'igual a']);
+
+  const [optionC] = useState([
+    'population', 'orbital_period', 'diameter', 'rotation_period', 'surface_water']);
+
+  function handleChange(event) {
+    const valueInput = event.target.value;
+    setFilter({
+      filterByName: {
+        name: valueInput,
+      },
+    });
+  }
+
+  function seleAChange(event) {
+    const { filterByNumericValues } = filters;
+    const seleAOptionSelected = event.target.value;
+    // console.log(seleAOptionSelected);
+    setFilter({ ...filters,
+      filterByNumericValues: { ...filterByNumericValues,
+        column: seleAOptionSelected,
+      },
+    });
+  }
+
+  function seleBChange(event) {
+    const { filterByNumericValues } = filters;
+    const seleBOptionSelected = event.target.value;
+    setFilter({ ...filters,
+      filterByNumericValues: { ...filterByNumericValues,
+        comparison: seleBOptionSelected,
+      },
+    });
+  }
+
+  function filtroNumberHandleChange(event) {
+    const { filterByNumericValues } = filters;
+    // console.log(event.target.value);
+    const filtroNumberHandleChangeSelected = event.target.value;
+    setFilter({ ...filters,
+      filterByNumericValues: { ...filterByNumericValues,
+        value: filtroNumberHandleChangeSelected,
+      },
+    });
+  }
+  // teste 2
+  function filterHandleClick() {
+    setClick(true);
+  }
+
+  const url = 'https://swapi-trybe.herokuapp.com/api/planets/';
+
+  // componentdidmount
+  // https://stackoverflow.com/questions/53332321/react-hook-warnings-for-async-function-in-useeffect-useeffect-function-must-ret
+  useEffect(() => {
+    async function fetchMyAPI() {
+      let data = await fetch(url);
+      data = await data.json();
```

Comment:
pra evitar o uso de let você pode reescrever essa duas linhas da seguinte forma
```suggestion
      const data = await fetch(url).then((response) => response.json());
```
desse modo modo você está esperando (await) diretamente pela execução da promise response.json() e assim ganha acesso direto ao data com o qual você esta trabalhando.